src/components/molecules/PostLinkEmbed/PostLinkEmbed.tsx

```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import * as Atoms from '@/atoms';
+import { useMemo } from 'react';
+
+export type PostLinkEmbedProps = {
+  content: string;
+};
+
+type LinkEmbed = 'none' | 'youtube';
+
+const parseContentForLinkEmbed = (content: string): { type: LinkEmbed; url: string } => {
+  try {
+    // Simple URL pattern for social media links
+    const urlRegex =
+      /(https?:\/\/[a-zA-Z0-9-._~:/?#[\]@!$&'()*+,;=%]+)|(?:www\.)?([a-zA-Z0-9][a-zA-Z0-9-]*\.)+[a-zA-Z]{2,}(?:\/[^\s]*)*/g;
```

Comment:
Problems:
- ReDoS vulnerability: This regex has nested quantifiers that could cause catastrophic backtracking with malicious input
- Overly permissive: Matches many things that aren't valid URLs
- Edge case: The regex allows URLs without protocols AND with protocols in the same pattern, making it complex.

Better approach: Use browser's built-in URL parser or a well-tested library

```suggestion
    // Match potential URLs more simply
    const urlRegex = /(?:https?:\/\/|www\.)[\S]+/gi;
```

Reply:
OK this is the same regex that is used the legacy app, I thought we wanted to keep the same functionality as there so that is why I used this parsing method (but improved other parts of the code). If we want to rethink all functionality, I agree that we can look into better options here. Was trying to follow the guidance of "consistent with the old app" and not change any core functionality, for example which types of links are handled in the new app vs old app.